superset-frontend/src/explore/components/controls/MetricControl/AdhocMetricEditPopover/index.jsx

```diff
@@ -370,7 +381,23 @@ export default class AdhocMetricEditPopover extends React.PureComponent {
               />
             </FormItem>
           </Tabs.TabPane>
-          <Tabs.TabPane key={EXPRESSION_TYPES.SIMPLE} tab={t('Simple')}>
+          <Tabs.TabPane
+            key={EXPRESSION_TYPES.SIMPLE}
+            tab={
+              extra.disallow_adhoc_metrics ? (
+                <Tooltip
+                  title={t(
+                    'Simple ad-hoc metrics are not enabled for this dataset',
```

Comment:
How about use the same message for both "SIMPLE" and "CUSTOM SQL":

```
Ad-hoc metrics are not supported by this dataset, use saved metrics instead.
```

Having different messages kind of means one can disallow SIMPLE and CUSTOM SQL separately.

Reply:
@ktmud the reason for the slightly different messages is the Druid NoSQL connector also disallows the `CUSTOM SQL` tab and thus I wanted to ensure consistency between those two experiences.

Reply:
I think these are two different scenarios. You can still use a different message for native Druid. IMO clarity is more important than consistency. If we have only one config value `disallow_ahoc_metrics`, then the message should corresponds to what this config value does----which isn't disabling SIMPLE and CUSTOM SQL separately.